### PR TITLE
Conversion of withdraw and txprepare to new UTXO primitives (as plugins!)

### PR DESCRIPTION
**Pull request by rustyrussell**
~On top of #3845~
~On top of #3958~

I've not touched fundchannel: it currently uses txprepare and friend and should instead use fundpsbt/utxopsbt, but I know @ZmnSCPxj wants to convert fundchannel to multifundchannel (Yes!).  Then txprepare etc should be deprecated.

But this shows just how much code we can get rid of from the wallet, which is nice.

### CODE REVIEW

**Comment by cdecker** [comment]
Converted to draft since #3845 is still work in progress.

**Comment by ZmnSCPxj** [comment]
We currently implement BIP69 via `common/permute_tx.c`.  Is there an equivalent API that works on PSBTs?  It looks like this PR has a regression, in that inputs and outputs are no longer sorted in BIP69 order (I think?)

**Comment by rustyrussell** [comment]
Yes, we preserve PSBT order, BUT:
1. change is added in random place.
2. fundpsbt selects UTXOs in random order.

Which I think is The Right Thing...

**Comment by ZmnSCPxj** [comment]
I would suggest shuffling the outputs --- you do not want to inadvertently leak any specific ordering the caller has, in case the caller does not shuffle.

Also, deserves a ChangeLog entry that we are no longer using BIP69 for withdrawal transactions.

**Comment by ZmnSCPxj** [comment]
Also `utxopsbt` does not shuffle inputs too.

**Comment by rustyrussell** [comment]
> Also `utxopsbt` does not shuffle inputs too.

No, that really is the caller's responsibility.  You might have sighash single sigs for example....

File: wallet/reservation.c [hunk 1]
```diff
@@ -208,6 +212,64 @@ static bool inputs_sufficient(struct amount_sat input,
...
+					   struct utxo **utxos,
+					   const struct ext_key *bip32_base,
+					   u32 nlocktime,
+					   u32 nsequence)
+{
+	struct pubkey key;
+	u8 *scriptSig, *scriptPubkey, *redeemscript;
+	struct bitcoin_tx *tx;
+
+	/* FIXME: Currently the easiest way to get a PSBT is via a tx */
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply wallet/reservation.c:224]: Thanks!  It's missing an nLocktime arg, so I've appended that to the series.

**Comment by rustyrussell** [comment]
> I would suggest shuffling the outputs --- you do not want to inadvertently leak any specific ordering the caller has, in case the caller does not shuffle.
> 
> Also, deserves a ChangeLog entry that we are no longer using BIP69 for withdrawal transactions.

Yes, there should be one:

    Changelog-Changed: JSON-RPC: `withdraw` now randomizes input and output order, not BIP69.
?

**Comment by rustyrussell** [comment]
OK, fixed the typo in the tests, appended a change to use create_psbt rather than via bitcoin_tx(), and rebased on top of #3958 which allows that.

**Comment by rustyrussell** [comment]
Trivial rebase, fix for python warning.

**Pull request merged**